packages/flutter/lib/src/services/text_input.dart

```diff
@@ -2624,6 +2644,9 @@ class SystemContextMenuController with SystemContextMenuClient, Diagnosticable {
 
   static SystemContextMenuController? _lastShown;
 
+  /// Map to store custom action callbacks, keyed by their unique identifiers.
+  static final Map<String, VoidCallback> _customActionCallbacks = <String, VoidCallback>{};
```

Comment:
If the method channel call were handled in binding.dart, then you could handle this stuff via SystemContextMenuClient and SystemContextMenuController.